test/distributed/test_c10d_nccl.py

```diff
@@ -1143,6 +1143,24 @@ def abortpg():
 
             thread.join()
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
```

Comment:
> given how python GC works and code holding references to the PG in multiple places, in practice calling destroy_process_group doesn't actually end up invoking the destructor

It would be great if we can add a test to verify this as well.

Reply:
Do you mean adding a test to validate `destroy_process_group` doesn't call PG destructor? That is tricky and non-deterministic since it depends on where references to the PG object are held in memory.

Reply:
One way to verify could be destroy the default WORLD and try to create WORLD again. If the resource was not freed, it would error out with the corresponding exception, IIRC.

Reply:
Do you have a case when we call `destroy_process_group`, PG destructor does not get called? Or this is not deterministic?

Reply:
Added a test.